listener/listener.go

```diff
@@ -0,0 +1,349 @@
+package listener
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Azure/azure-amqp-common-go/v3/auth"
+	servicebus "github.com/Azure/azure-service-bus-go"
+	"github.com/keikumata/azure-pub-sub/message"
+
+	servicebusinternal "github.com/keikumata/azure-pub-sub/internal/servicebus"
+)
+
+const (
+	defaultSubscriptionName = "default"
+)
+
+// Listener is a struct to contain service bus entities relevant to subscribing to a publisher topic
+type Listener struct {
+	namespace          *servicebus.Namespace
+	topicEntity        *servicebus.TopicEntity
+	subscriptionEntity *servicebus.SubscriptionEntity
+	listenerHandle     *servicebus.ListenerHandle
+	topicName          string
+	subscriptionName   string
+	filterDefinitions  []*filterDefinition
+}
+
+// Subscription returns the servicebus.SubscriptionEntity that the listener is setup with
+func (l *Listener) Subscription() *servicebus.SubscriptionEntity {
+	return l.subscriptionEntity
+}
+
+// Topic returns servicebus.TopicEntity that the listener is setup with
+func (l *Listener) Topic() *servicebus.TopicEntity {
+	return l.topicEntity
+}
+
+// Namespace returns the servicebus.Namespace that the listener is setup with
+func (l *Listener) Namespace() *servicebus.Namespace {
+	return l.namespace
+}
+
+// ManagementOption provides structure for configuring a new Listener
+type ManagementOption func(l *Listener) error
+
+// Option provides structure for configuring when starting to listen to a specified topic
+type Option func(l *Listener) error
+
+// WithConnectionString configures a listener with the information provided in a Service Bus connection string
+func WithConnectionString(connStr string) ManagementOption {
```

Comment:
all option funcs are stripped from the Listener prefix. they are already in the listener package